crates/types/src/v1alpha1/wasmcloud_host_config.rs

```diff
@@ -187,6 +189,36 @@ fn default_nats_leafnode_port() -> u16 {
     7422
 }
 
+#[derive(Deserialize, Serialize, Clone, Debug, JsonSchema)]
+pub struct ConfigMapSource {
+    /// Name of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
+    pub name: String,
+
+    /// Specify whether the ConfigMap must be defined
+    pub optional: Option<bool>,
+}
+
+#[derive(Deserialize, Serialize, Clone, Debug, JsonSchema)]
+pub struct SecretSource {
+    /// Name of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
+    pub name: String,
+
+    /// Specify whether the ConfigMap must be defined
+    pub optional: Option<bool>,
+}
+
+#[derive(Deserialize, Serialize, Clone, Debug, JsonSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct CertificateSource {
```

Comment:
modelled after https://pkg.go.dev/k8s.io/api/core/v1#EnvFromSource

Thinking we should support:
- configmaps
- secrets
- hostPaths